src/helpers/isInsidePolygon.js

```diff
@@ -1,25 +1,25 @@
-const isInsidePolygon = async (lat, long, k, polygons, coordinates) => {
-  
+const isInsidePolygon = (lat, long, k, polygons, coordinates) => {
+
   const nVert = polygons[k].length;
   const testY = lat;
   const testX = long;
 
-  let vertY;
-  let vertX;
+  let vertY = [];
+  let vertX = [];
 
   for (let t = 0; t < nVert; t++) {
-    vertY = coordinates[polygons[k][t] - 1][0];
-    vertX = coordinates[polygons[k][t] - 1][1];
+    vertY.push(coordinates[polygons[k][t] - 1][0]);
+    vertX.push(coordinates[polygons[k][t] - 1][1]);
   }
 
-  let c = 0;
-  const helper = vertY[i] > testY !== vertY[j] > testY && testX < ((vertX[j] - vertX[i]) * (testY - vertY[i])) / (vertY[j] - vertY[i]) + vertX[i];
-
   for (let i = 0, j = nVert - 1; i < nVert; j = i++) {
-    if (helper) c = !c;
+    if (((vertY[i] > testY) !== (vertY[j] > testY)) &&
+      (testX < (vertX[j] - vertX[i]) * (testY - vertY[i]) / (vertY[j] - vertY[i]) + vertX[i])) {
+      return true;

```

Comment:
Are you sure? Only one validation is necessary?

Reply:
in back of php this is the only validation, it is validated if the polygon is inside using the coordinates obtained previously. In fact, in the code I only changed that we initialized the helper before the for loop and it gave an error because it wanted to access a variable i that did not exist before declaring that loop

Reply:
Ok.